workflows_core/engine/stable_engine.py

```diff
@@ -14,18 +15,36 @@ def __init__(self, *args, **kwargs):
 
         self._show_progress_bar = kwargs.pop("show_progress_bar", True)
 
-    def apply(self) -> Any:
+    def apply(self) -> None:
+        """
+        Returns the ratio of successful chunks / total chunks needed to iterate over the dataset
+        """
 
         iterator = self.iterate()
+        successful_chunks = 0
+        error_logs = []
 
         for chunk in tqdm(
             iterator,
             desc=repr(self.operator),
             disable=(not self._show_progress_bar),
             total=self.num_chunks,
         ):
-            new_batch = self.operator(chunk)
-            result = self.update_chunk(new_batch)
-            logging.debug(result)
-
-        return
+            try:
+                new_batch = self.operator(chunk)
+            except Exception as e:
+                chunk_error_log = {
+                    "exception": str(e),
+                    "traceback": traceback.format_exc(),
```

Comment:
We can't store tracebacks - just the exception.

Reply:
this is the traceback formatted as a string, are you sure this not okay

Reply:
Yes. imagine looping through 400 chunks and each of them storing 200 lines of traceback. You'll only be able to run 1 workflow before this starts having problems.

Instead - can we store the traceback of only the last failed chunk? 
This will be what's required in 90% of cases.

Reply:
Okay - I was told each workflow was 16MB - we can do this not a problem.